.github/workflows/upstream.yml

```diff
@@ -9,13 +9,30 @@ jobs:
   versioncheck:
     runs-on: ubuntu-latest
     if: github.repository_owner == 'daviddrysdale'
+    permissions:
+      issues: write
     steps:
       - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3
       - uses: actions/setup-python@d27e3f3d7c64b4bbf8e4abfb9b63b83e846e0435 # v4
         with:
           python-version: 3
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install requests
+
       - name: Check upstream version
         run: |
-          pip install -U pip
-          pip install requests
-          python3 tools/python/checkupstreamversion.py
+          python tools/python/checkupstreamversion.py
+
+      - name: Create update issue
+        if: failure()
+        uses: actions/github-script@d7906e4ad0b1822421a7e6a35d5ca353c962f410  # v6.4.1
+        with:
+          # language=JavaScript
+          script: |
+            await github.rest.issues.create({
```

Comment:
Will this create a new issue on every failed run of the workflow?  

I'm thinking that if I'm away for a week and a release happens to come out at the start, would I return to 14 identical issues? :-)